### PR TITLE
Add backstage config

### PR DESCRIPTION
**Pull request by m4mm4r**
If approved, this PR will:

- Add `.backstage` directory, with config for existing auth and book APIs.

### CODE REVIEW

File: .backstage/api-book-grpc.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  name: book-grpc
+  description: |
+    Einride Extend: Shipment API (gRPC)
+spec:
+  type: grpc
+  lifecycle: production
+  owner: team-transportation
+  system: extend
+  definition:
+    $text: ../proto/einride/saga/extend/book/v1beta1/booking_service.proto
```
Comment (m4mm4r) [inline .backstage/api-book-grpc.yaml:13]: This won't cover the `shipment_service.proto` docs, happy to take suggestions on how to get that in as well.

Comment (odsod) [reply .backstage/api-book-grpc.yaml:13]: I think the principle is to register each gRPC service as it's own API

Comment (m4mm4r) [reply .backstage/api-book-grpc.yaml:13]: Alright then it won't be 1-1 with the openapi structure but maybe there's no problem if it isn't!

Comment (odsod) [reply .backstage/api-book-grpc.yaml:13]: For now that should be OK, we can make "breaking changes" to the software catalog - but let's take this as i sign that we want to develop principles around how we structure our public gRPC APIs to ensure cohesiveness between the gRPC and HTTP APIs.

**Review by odsod — APPROVED** [review]

**Pull request merged**